tracing_test.go

```diff
@@ -0,0 +1,315 @@
+			want = "non-" + want
+		}
+		t.Errorf("got TraceID = %s, want %s", span.TraceID, want)
+	}
+	if got := span.SpanID == zeroSpanID; got != c.ZeroSpanID {
+		want := "zero"
+		if !c.ZeroSpanID {
+			want = "non-" + want
+		}
+		t.Errorf("got SpanID = %s, want %s", span.SpanID, want)
+	}
+	if got, want := span.Sampled, c.Sampled; got != want {
+		t.Errorf("got Sampled = %v, want %v", got, want)
+	}
+
+	if got, want := len(span.spanRecorder().spans), c.RecorderLen; got != want {
+		t.Errorf("got %d spans in recorder, want %d", got, want)
+	}
+
+	if span.StartTime.IsZero() {
+		t.Error("start time not set")
+	}
+	if span.EndTime.IsZero() {
+		t.Error("end time not set")
+	}
+	if span.EndTime.Before(span.StartTime) {
+		t.Error("end time before start time")
+	}
+}
+
+func TestToSentryTrace(t *testing.T) {
+	tests := []struct {
+		span *Span
+		want string
+	}{
+		{&Span{}, "00000000000000000000000000000000-0000000000000000"},
+		{&Span{Sampled: SampledTrue}, "00000000000000000000000000000000-0000000000000000-1"},
+		{&Span{Sampled: SampledFalse}, "00000000000000000000000000000000-0000000000000000-0"},
+		{&Span{TraceID: TraceID{1}}, "01000000000000000000000000000000-0000000000000000"},
+		{&Span{SpanID: SpanID{1}}, "00000000000000000000000000000000-0100000000000000"},
+	}
+	for _, tt := range tests {
+		if got := tt.span.ToSentryTrace(); got != tt.want {
+			t.Errorf("got %q, want %q", got, tt.want)
+		}
+	}
+}
+
+func TestContinueSpanFromRequest(t *testing.T) {
+	traceID := TraceIDFromHex("bc6d53f15eb88f4320054569b8c553d4")
+	spanID := SpanIDFromHex("b72fa28504b07285")
+
+	for _, sampled := range []Sampled{SampledTrue, SampledFalse, SampledUndefined} {
+		sampled := sampled
+		t.Run(sampled.String(), func(t *testing.T) {
+			var s Span
+			hkey := http.CanonicalHeaderKey("sentry-trace")
+			hval := (&Span{
+				TraceID: traceID,
+				SpanID:  spanID,
+				Sampled: sampled,
+			}).ToSentryTrace()
+			header := http.Header{hkey: []string{hval}}
+			ContinueFromRequest(&http.Request{Header: header})(&s)
+			if s.TraceID != traceID {
+				t.Errorf("got %q, want %q", s.TraceID, traceID)
+			}
+			if s.ParentSpanID != spanID {
+				t.Errorf("got %q, want %q", s.ParentSpanID, spanID)
+			}
+			if s.Sampled != sampled {
+				t.Errorf("got %q, want %q", s.Sampled, sampled)
+			}
+		})
+	}
+}
+
+func TestSpanFromContext(t *testing.T) {
+	// SpanFromContext always returns a non-nil value, such that you can use
+	// it without nil checks.
+	// When no span was in the context, the returned value is a no-op.
+	// Calling StartChild on the no-op creates a valid transaction.
+	// SpanFromContext(ctx).StartChild(...) === StartSpan(ctx, ...)
+
+	ctx := NewTestContext(ClientOptions{})
+	span := spanFromContext(ctx)
+
+	_ = span
+
+	// SpanCheck{
+	// 	ZeroTraceID: true,
+	// 	ZeroSpanID:  true,
+	// }.Check(t, span)
+
+	// // Should create a transaction
+	// child := span.StartChild("top")
+	// SpanCheck{
+	// 	RecorderLen: 1,
+	// }.Check(t, child)
```

Comment:
This test is just a stub for `SpanFromContext`. Contents are commented out because `spanFromContext` doesn't return a "good enough dummy" to allow for the flow described by the test.

In a future pass through the tracing implementation, we either remove `spanFromContext` and everything related to it, or have to implement it such to make this test pass.